### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by maksat990rd**

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import kg.peaksoft.giftlistb6.db.services.UserProfileService;
+import kg.peaksoft.giftlistb6.dto.requests.ProfileRequest;
+import kg.peaksoft.giftlistb6.dto.responses.FriendProfileResponse;
+import kg.peaksoft.giftlistb6.dto.responses.ProfileResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/userInfo")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java]: Maybe api/profile

File: src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public class UserInfoApi {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java]: Add cross-origin, swagger descriptions, and security permission for user

Comment (i-Coderr) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java]: Where is the user's own profile endpoint?

**Comment by i-Coderr** [comment]
First send to code review

**Comment by i-Coderr** [comment]
Resolve conflicts

File: src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java (hunk 1 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java:13]: +

Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserInfoApi.java:13]: Ready

**Comment by Aiza-anarbekova** [comment]
не хватает одного метода

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDate;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class HolidayResponses2 {
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java:14]: HolidayResponses2 ??? Please write the name properly for the class/methods

File: src/main/resources/application-max.properties [hunk 4]
```diff
@@ -0,0 +1,41 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/giftlist
```
Comment (i-Coderr) [inline src/main/resources/application-max.properties]: Don't add your config files

File: src/main/resources/application-max.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+spring:
```
Comment (i-Coderr) [inline src/main/resources/application-max.yml]: WTF?

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    private Long id;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java:15]: Write to the next line

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java (hunk 3 above)
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/HolidayResponses2.java:14]: HolidayResponses2 ??? Please write the name properly for the class/methods

**Review by i-Coderr — COMMENTED** [review]
Don not add your application config files to the pull request

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+@RequiredArgsConstructor
+@RequestMapping("api/profile")
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "User profile api", description = "User can save,update profile")
+@PreAuthorize("hasAuthority('USER')")
+public class UserProfileApi {
+
+    private final UserProfileService service;
+
+    @PostMapping
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: @PostMapping же башка маппингтер методдун үстүнө жазылат

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import kg.peaksoft.giftlistb6.dto.responses.ProfileResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/profile")
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "User profile api", description = "User can save,update profile")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: Write just Profile API only

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+@PreAuthorize("hasAuthority('USER')")
+public class UserProfileApi {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: ProfileApi

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    private final UserProfileService service;
+
+    @PostMapping
+    @Operation(summary = "Save profile", description = "user can save profile  ")
+    public ProfileResponse saveUserInfo(@RequestBody ProfileRequest request)  {
+        return service.saveProfile(request);
+    }
+
+    @PutMapping("/{id}")
+    @Operation(summary = "update profile", description = "user can update profile")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: u should be capital

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+@RequestMapping("api/profile")
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "User profile api", description = "User can save,update profile")
+@PreAuthorize("hasAuthority('USER')")
+public class UserProfileApi {
+
+    private final UserProfileService service;
+
+    @PostMapping
+    @Operation(summary = "Save profile", description = "user can save profile  ")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: u should be capital

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @PutMapping("/{id}")
+    @Operation(summary = "update profile", description = "user can update profile")
+    public ProfileResponse updateProfileUser(@PathVariable Long id, @RequestBody ProfileRequest request) {
+        return service.saveUpdateUser(id,request);
+    }
+
+    @GetMapping("{id}")
+    @Operation(summary = "My friend profile", description = "user can see friend profile")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: u should be capital

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    public FriendProfileResponse FriendProfile(@PathVariable Long id){
+        return service.friendProfile(id);
+    }
+
+    @GetMapping("/show/{id}")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: should be only "me"

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    @Operation(summary = "My profile", description = "user can see profile")
+    public FriendShowResponse myProfile(@PathVariable Long id) {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: Why responding FriendShowResponse?

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java (hunk 7 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: сделано

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java (hunk 8 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: готово

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java (hunk 9 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: готово

File: src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java (hunk 14 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/UserProfileApi.java]: Исправлено

File: src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Operation(summary = "Friends profile", description = "Can see friends profile")
+    @GetMapping("{id}")
+    public FriendProfileResponse FriendProfile(@PathVariable Long id) {
+        return service.friendProfile(id);
+    }
+
+    @Operation(summary = "My profile", description = "Can see profile")
+    @GetMapping("/me/{id}")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java]: me эле болот, {id} кереги жок

File: src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+import kg.peaksoft.giftlistb6.dto.responses.ProfileResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/profile")
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Profile api", description = "Can save, update profile")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java]: Please write Profile API

File: src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Operation(summary = "Friends profile", description = "Can see friends profile")
+    @GetMapping("{id}")
+    public FriendProfileResponse FriendProfile(@PathVariable Long id) {
+        return service.friendProfile(id);
+    }
+
+    @Operation(summary = "My profile", description = "Can see profile")
+    @GetMapping("/me/{id}")
+    public MyProfileResponse myProfile(@PathVariable Long id) {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java]: Delete (@PathVariable Long id) in the method

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/MyProfileResponse.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class MyProfileResponse {
+
+    private Long id;
+    private String photo;
+    private String fullName;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/MyProfileResponse.java]: add firstName and lastName not fullName

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/MyProfileResponse.java [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+    private String email;
+}
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/MyProfileResponse.java:16]: add userInfo

**Review by i-Coderr — COMMENTED** [review]
Don't add your application.properties files in the pull request!

File: src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Operation(summary = "Friends profile", description = "Can see friends profile")
+    @GetMapping("{id}")
+    public FriendProfileResponse FriendProfile(@PathVariable Long id) {
+        return service.friendProfile(id);
+    }
+
+    @Operation(summary = "My profile", description = "Can see profile")
+    @GetMapping
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/ProfileApi.java]: add **me**

**Pull request merged**